src/spatial_ref/srs.rs

```diff
@@ -0,0 +1,171 @@
+use libc::{c_int, c_char, c_void};
+use std::ffi::{CString};
+use std::ptr;
+use utils::_string;
+use gdal_sys::{osr, ogr_enums};
+
+use errors::*;
+
+pub struct CoordTransform(*mut c_void);
+
+impl Drop for CoordTransform {
+    fn drop(&mut self) {
+        unsafe { osr::OCTDestroyCoordinateTransformation(self.0) };
+        self.0 = ptr::null_mut();
+    }
+}
+
+impl CoordTransform {
+    pub fn new(sp_ref1: &SpatialRef, sp_ref2: &SpatialRef) -> Result<CoordTransform> {
+        let c_obj = unsafe { osr::OCTNewCoordinateTransformation(sp_ref1.0, sp_ref2.0) };
+        if c_obj.is_null() {
+            return Err(ErrorKind::NullPointer("OCTNewCoordinateTransformation").into());
+        }
+        Ok(CoordTransform(c_obj))
+    }
+
+    pub fn transform_coord(&self, x: &mut [f64], y: &mut [f64], z: &mut [f64]){
+        let nb_coords = x.len();
+        assert_eq!(nb_coords, y.len());
+        let ret_val = unsafe { osr::OCTTransform(
+            self.0,
+            nb_coords as c_int,
+            x.as_mut_ptr(),
+            y.as_mut_ptr(),
+            z.as_mut_ptr()
+        ) };
+        assert_eq!(true, ret_val);
+    }
+
+    pub fn to_c_hct(&self) -> *const c_void {
+        self.0 as *const c_void
+    }
+}
+
+pub struct SpatialRef(*mut c_void);
+
+impl Drop for SpatialRef {
+    fn drop(&mut self){
+        unsafe { osr::OSRDestroySpatialReference(self.0)};
+        self.0 = ptr::null_mut();
+    }
+}
+
+impl Clone for SpatialRef {
+    fn clone(&self) -> SpatialRef {
+        let n_obj = unsafe { osr::OSRClone(self.0 as *const c_void)};
+        SpatialRef(n_obj)
+    }
+}
+
+impl PartialEq for SpatialRef {
+    fn eq(&self, other: &SpatialRef) -> bool {
+        unsafe { osr::OSRIsSame(self.0, other.0)}
+    }
+}
+
+impl SpatialRef {
+    pub fn new() -> Result<SpatialRef> {
+        let c_obj = unsafe { osr::OSRNewSpatialReference(ptr::null()) };
+        if c_obj.is_null() {
+            return Err(ErrorKind::NullPointer("OSRNewSpatialReference").into());
+        }
+        Ok(SpatialRef(c_obj))
+    }
+
+    pub fn new_from_definition(definition: &str) -> Result<SpatialRef> {
```

Comment:
This is indeed a significant list, thx (by looking around i also came across the RFC 430 which mention this `from_some_other_type` kind of constructor).